source/frontend/src/features/mapSideBar/compensation/update/UpdateCompensationRequisitionForm.tsx

```diff
@@ -264,7 +264,9 @@ const UpdateCompensationRequisitionForm: React.FC<CompensationRequisitionFormPro
                       selectFunction={(optionPayees, selectedPayees) =>
                         optionPayees.filter(payee =>
                           selectedPayees?.find(
-                            selectedPayee => selectedPayee.api_id === payee.api_id,
```

Comment:
The old logic was selecting some payees when they shouldn't. If you had an owner and an interest holder that had the same id then the multiselect would show **both** when one of them was selected. This happens because the same Id number (ie 1) can be used in different tables that feed into the payees multi-select.

Fixed by also checking for the type of payee (Owner, InterestHolder, etc) in addition to the entity key